### DISCUSSION TITLE
Create an ontology from which derive the Yaml format

### DISCUSSION THREAD

**Discussion by EmidioStani** [publiccodeyml/publiccode.yml#133] [Ideas]
It is a really an interesting project, discovered today at Fosdem, I wonder if there is an ontology or a vocabulary created (I see there are definitions, cardinalities, etc) in RDF from which derive the Yaml format.

Since I work for the SEMICeu project, where they are vocabularies like CPOV (which defines a Public Organisation which in turn could use a software described in public code), it could be nice linking them.

**Comment by ruphy** [comment]
Dear Emidio,

thanks for your comment! As this is an international forum, would you mind posting your comment in English? Thank you! 😄

**Reply by EmidioStani** [reply]
Updated description, sorry

**Comment by bfabio** [comment]
Hello @EmidioStani there's no such thing yet, would you like to take a shot at it? :)

**Comment by bfabio** [comment]
cc @mfortini

**Comment by nichtich** [comment]
I think the better approach would be the other way round: create a mapping from the Yaml-format to RDF. Best method to do so is JSON-LD (or YAML-LD, practically the same). A starting point to find RDF properties that publiccode.yml can be mapped to is [Description of a Project (DOAP)](https://en.wikipedia.org/wiki/DOAP) vocabulary.

To formally define the Yaml format, the right level of description is not RDF but a Schema language such as JSON Schema (can be used [for YAML as well](https://json-schema-everywhere.github.io/yaml))
